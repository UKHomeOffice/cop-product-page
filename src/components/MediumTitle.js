import React from "react"

const MediumTitle = (props) => (
  <div className="govuk-heading-m">
    {props.text}
  </div>
);

export default MediumTitle