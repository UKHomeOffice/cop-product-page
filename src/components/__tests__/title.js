import React from "react";
import renderer from "react-test-renderer";
import ExtraLargeTitle from "../extra-large-title";
import LargeTitle from "../large-title";
import MediumTitle from "../MediumTitle";

describe("title", () => {
  it("renders extra large title correctly", () => {
    const tree = renderer.create(<ExtraLargeTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Returns the correct extra large title 'text' prop passed in", () => {
    const tree = renderer.create(<ExtraLargeTitle text={"a title"} />);
    const testInstance = tree.root;
    expect(testInstance.props.text).toBe("a title");
  });

  it("renders large title correctly", () => {
    const tree = renderer.create(<LargeTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Returns the correct large title 'text' prop passed in", () => {
    const tree = renderer.create(<LargeTitle text={"a title"} />);
    const testInstance = tree.root;
    expect(testInstance.props.text).toBe("a title");
  });

  it("renders medium title correctly", () => {
    const tree = renderer.create(<MediumTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Returns the correct medium title 'text' prop passed in", () => {
    const tree = renderer.create(<MediumTitle text={"a title"} />);
    const testInstance = tree.root;
    expect(testInstance.props.text).toBe("a title");
  });
});
