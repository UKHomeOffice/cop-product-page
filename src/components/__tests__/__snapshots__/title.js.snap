// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title renders extra large title correctly 1`] = `
<h1
  className="govuk-heading-xl"
/>
`;

exports[`title renders large title correctly 1`] = `
<h2
  className="govuk-heading-l"
/>
`;

exports[`title renders medium title correctly 1`] = `
<h3
  className="govuk-heading-m"
/>
`;
